src/pages/HomePage/components/TrendingGifts/ProductCard/ProductCard.tsx

```diff
@@ -7,19 +8,42 @@ import {
   ProductName,
   ProductPrice,
   RankBadge,
-} from './ProductCard.styles';
+} from "./ProductCard.styles";
+import { useAuth } from "@/contexts/AuthContext";
+import { ROUTES } from "@/constants/routes";
 
 export interface ProductCardPropsType {
+  id: number;
   imageURL: string;
   name: string;
   brandName: string;
   sellingPrice: number;
   index: number;
 }
 
-function ProductCard({ imageURL, name, brandName, sellingPrice, index }: ProductCardPropsType) {
+function ProductCard({
+  id,
+  imageURL,
+  name,
+  brandName,
+  sellingPrice,
+  index,
+}: ProductCardPropsType) {
+  const { isLoggedIn } = useAuth();
+  const navigate = useNavigate();
+
+  const handleClick = () => {
+    const redirectPath = ROUTES.ORDER.replace(":id", id.toString());
+    if (isLoggedIn) {
+      navigate(redirectPath);
+      return;
+    }
+
+    navigate(`${ROUTES.LOGIN}?redirect=${encodeURIComponent(redirectPath)}`);
+  };
```

Comment:
이렇게 하면 코드가 더 간결해질 수 있을 것 같아요!
```suggestion
navigate(isLoggedIn ? '' : '')
```

Reply:
좋은 제안 감사합니다! 다만 개인적으로는 현재 방식이 두 가지 다른 네비게이션 로직을 명확하게 구분해서 가독성 면에서 더 좋다고 느꼈는데, 리뷰어님의 의견이 궁금합니다!

특히 나중에 각 분기에 추가 로직이 있을 경우를 생각하면, if-else로 분리해두는 게 확장성 측면에서도 좋을 것 같다는 생각이 들었습니다.

혹시 다른 관점이나 고려사항이 있으시면 알려주시면 감사하겠습니다.